yt_dlp/extractor/generic.py

```diff
@@ -3708,6 +3724,12 @@ def filter_video(urls):
             if embed_url and embed_url != url:
                 return self.url_result(embed_url)
 
+        if not found:
+            # Facebook external link warning URL that doesn't auto-redirect
+            video_url = next(iter(re.findall(r'^https?://(?:www\.)?facebook\.com/flx/warn/\?u=([^&]+)', url)), None)
+            if video_url:
+                return self.url_result(compat_urllib_parse_unquote(video_url))
+
```

Comment:
since it is matching by URL, why not make an extractor for this?

Reply:
ignoring that `iter()` wouldn't be needed for now, why do we need an extractor just for redirect page?

Reply:
Seems a little overkill to me to make every such indirect redirection page that may come up its own extractor, since the URL it gets has to be passed on to generic anyway as it can be anything.

Reply:
but now we are downloading the webpage, though not doing anything with it

Reply:
since it doesn't require webpage content, it may be moved here https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/generic.py#L2497

Reply:
I still think a new extractor is better, but this is an acceptable solution